src/content/basics/ha-masters/index.md

```diff
@@ -0,0 +1,169 @@
+
+When high availability master nodes is activated for a cluster, the three
+master nodes are spread over different availability zones (AZ). Depending on
+the number of AZ in the region, the logic is:
+
+- With **three or more AZs** in the region, each master node is assigned to a different AZ, selected randomly.
+- With **two AZs** in the region (as is the case in `cn-north-1`), both AZs get used. Two master nodes will share the same AZ.
+
+When [converting a single master cluster to high availability](#conversion-to-ha),
+the AZ used by the master node before is re-used. Additional AZs are assigned
+by applying the logic described above. Here, the AZ assignment of existing
+worker [node pools](/basic/nodepools/) is taken into account.
+
+## Upgrades from previous releases {#upgrades}
+
+When upgrading a cluster to release v11.4.0, the cluster will remain a single
+master cluster during and after the upgrade. The API unavailability during the
+upgrade that is typical for single master clusters will apply for this upgrade.
+
+A conversion to high availability can be triggered manually after the upgrade
+is finished.
+
+## Conversion from single master to high availability {#conversion-to-ha}
+
+Single master clusters using release v{{% first_aws_ha_masters_version %}} or
+above on AWS can be converted to master node high availability in the user
+interfaces and via the APIs.
+
+## Via the web UI {#web-ui}
+
+The web UI presents information on the master node in the cluster details page.
+Next to these details you find a button _Switch to high availability…_, unless
+the cluster is currently undergoing an upgrade. Click this button and follow
+the instructions in the web UI.
+
+## Via the CLI (`gsctl`) {#gsctl}
+
+The `gsctl` CLI provides the [gsctl update cluster](/reference/gsctl/update-cluster/) to change cluster details. Check the reference for the `--master-ha` flag.
+
+## Via the Rest API {#rest-api}
+
+Check the [v5 cluster modification API reference](/api/#operation/modifyClusterV5) to find out how to convert a cluster programmatically using the Rest API.
+
+## Via the Control Plane K8s API {#cp-k8s-api}
+
+Two resources have to be changed in order to convert a single master cluster
+to high availability.
+
+1. `AWSControlPlane`
+2. `G8sControlPlane`
+
+### `AWSControlPlane`
+
+The cluster's [`AWSControlPlane`](/reference/cp-k8s-api/awscontrolplanes.infrastructure.giantswarm.io/)
+has a `.spec.availabilityZones` array which, in case of a single master
+cluster, has only one item. This item is the availability zone assigned to the
+only master node.
+
+To convert the cluster to high availability, this array has to be extended
+to have three items. Each item specifies the availability zone for one master
+node after the conversion.
+
+**CAUTION:** Order is important here. The first item must be left untouched. Or, in other words, what was the only array item before must be the first of three item after the change.
```

Comment:
@anvddriesch @paurosello is this true? Just asking because I am not into the details of defaulting. I was in general confused because in the operator we sort the AZs before using them so the order shouldn't make a big difference.

Reply:
It is my understanding of this thread: https://gigantic.slack.com/archives/C0148R1G12N/p1592236645153400

Reply:
Then we need to test this, order is important I think

Reply:
I'll simplify this so that only the `G8sControlPlane` change is mentioned, to leave the `AWSControlPlane` change to automation.